bin/proving-service/src/proxy/health_check.rs

```diff
@@ -38,49 +30,17 @@ impl BackgroundService for LoadBalancerState {
                 // Create a new spawn to perform the health check
                 let span = debug_span!("proxy:health_check");
                 let _guard = span.enter();
+                {
+                    let mut workers = self.workers.write().await;
```

Comment:
Let's create an issue to figure out how to avoid locking the entire workers vector while health checks are happening.

Reply:
We already track this in: https://github.com/0xPolygonMiden/miden-base/issues/1276